### ISSUE TITLE
ansible-core 2.13 ignores error in import in template

### ISSUE THREAD

**Issue by user318** [ansible/ansible#78156]
### Summary

When I do {% import %} in some jinja template and that imported template contains errors or raises an exception, it is just ignored. Before 2.13 I saw the error when running the playbook with the exception from the imported template. With 2.13 it is just ignored and everything is green. At least such mistakes are ignored: `{{ _undefined_name }}`, `{{ {}['missing_attribute'] }}`, but division on zero still yields an error: `{{ 0/0 }}`.
I tried different combinations of ansible 2.12 & 2.13 with jinja2 3.0 & 3.1. The problem arises only with ansible 2.13 with both jinja versions.

### Issue Type

Bug Report

### Component Name

jinja2

### Ansible Version

```console
$ ansible --version
ansible [core 2.13.1]
  config file = /home/user/.ansible.cfg
  configured module search path = ['/home/user/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/user/soft/ansible-test-2.13.1/lib/python3.10/site-packages/ansible
  ansible collection location = /home/user/.ansible/collections:/usr/share/ansible/collections
  executable location = ./bin/ansible
  python version = 3.10.5 (main, Jun  8 2022, 02:00:39) [GCC 10.2.1 20201203]
  jinja version = 3.1.2
  libyaml = True
```

### Configuration

```console
# if using a version older than ansible-core 2.12 you should omit the '-t all'
$ ansible-config dump --only-changed -t all
```

### OS / Environment

Void Linux

### Steps to Reproduce

```yaml (paste below)
---

- hosts: localhost
  connection: local
  tasks:

  - name: test
    template:
      dest: "test.txt"
      src: test1.j2
```

test1.j2:
```
{% import 'test2.j2' as t %}
```

test2.j2:
```
{{ _error }}
{{ {}["error"] }}
```

### Expected Results

With ansible 2.12 I get an error:
```
TASK [test] ********************************************************************
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: ansible.errors.AnsibleUndefinedVariable: '_error' is undefined
fatal: [localhost]: FAILED! => {"changed": false, "msg": "AnsibleUndefinedVariable: '_error' is undefined"}
```

### Actual Results

```console
$ ansible-playbook -CD test.yml 
[WARNING]: No inventory was parsed, only implicit localhost is available
[WARNING]: provided hosts list is empty, only localhost is available. Note that
the implicit localhost does not match 'all'

PLAY [localhost] ***************************************************************

TASK [Gathering Facts] *********************************************************
ok: [localhost]

TASK [test] ********************************************************************
--- before
+++ after: /home/user/.ansible/tmp/ansible-local-17138l0v8fvxh/tmpla8gx7_o/test1.j2
@@ -0,0 +1 @@
+

changed: [localhost]

PLAY RECAP *********************************************************************
localhost                  : ok=2    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

**Comment by ansibot** [comment]
Files identified in the description:
* `lib/ansible/template`

If these files are incorrect, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/devel/ISSUE_HELP.md)

**Comment by mkrizek** [comment]
Since `t` is unused should it fail? One could argue the imported template is evaluated lazily, when it comes to undefined values anyway.

**Comment by user318** [comment]
> Since `t` is unused should it fail? One could argue the imported template is evaluated lazily, when it comes to undefined values anyway.

In this simplified example - actually yes. But I have noticed this behaviour with my production templates, I used such undefined errors to see some clue in error message, when I check some conditions in the template. And after upgrading to 2.13 those errors become invisible.
I tried to use the imported template in this simple case and I have the errors about undefined variables now, so it looks like some more complex example is needed here. I'll try to improve it.
